templates/pulpcore-api.service.erb

```diff
@@ -8,9 +8,10 @@ Environment="DJANGO_SETTINGS_MODULE=pulpcore.app.settings"
 Environment="PULP_SETTINGS=<%= scope['pulpcore::settings_file'] %>"
 Environment="PULP_STATIC_ROOT=<%= scope['pulpcore::pulp_static_root'] %>"
 User=<%= scope['pulpcore::user'] %>
-PIDFile=/run/pulpcore-api.pid
+Group=<%= scope['pulpcore::group'] %>
+WorkingDirectory=%t/pulpcore-api
```

Comment:
I *think* %t resolves to `/run` when it runs as a system service. I'm a bit unsure about this. I've considered setting it to `~` or explicit `/var/lib/pulp`. That way systemd also automatically ensures `/var/lib/pulp` is present and mounted. I'm not sure why these services should have a different workdir.

Reply:
I'm unclear from the comment what we want to end up with. If `/var/lib/pulp` is the proper working directory then I vote set it explicitly to that.

Reply:
I think `%t` resolves to `/run` for system services. This is to preserve the previous `/var/run` (and on all modern systems that's a symlink to `/run`. I'm leaning to `~` or `/var/lib/pulp` because systemd will ensure it's mounted which can be very useful if it comes from NFS or a separate partition (which is very common), but I might missing something.

Here I'd like some feedback from Pulp upstream since I don't know what their preference is.

Reply:
On Pulp upstream, we never considered the `$PWD` of the pulp processes.

I agree with `~` (pulp_installer defaults the pulp  home dir to `/var/lib/pulp`, but lets users change it), in case we do end up writing anything there, and so that it gets mounted.

Reply:
> (pulp_installer defaults the pulp home dir to `/var/lib/pulp`, but lets users change it)

So does this module, but since it also manages the user, the homedir will be updated and that causes service restarts. Technically I think that Linux doesn't allow changing the homedir if a user is logged in, so you need to manually stop all services before, but after that it should work. Of course there's no data migration either.